api/test_history_test.go

```diff
@@ -0,0 +1,77 @@
+//go:build medium
+// +build medium
+
+// Copyright 2023 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package api
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"github.com/web-platform-tests/wpt.fyi/shared/sharedtest"
+)
+
+func TestHistoryHandler(t *testing.T) {
+	ctx, done, err := sharedtest.NewAEContext(true)
+	assert.Nil(t, err)
+	defer done()
+
+	sampleRun := shared.TestHistoryEntry{
+		BrowserName: "chrome",
+		RunID:       "123",
+		Date:        "2022-06-02T06:02:55.000Z",
+		TestName:    "test name",
+		SubtestName: "",
+		Status:      "PASS",
+	}
+
+	body :=
+		`{
+			"testName": "test name"
```

Comment:
the naming convention for payload should be [snake case](https://github.com/web-platform-tests/wpt.fyi/tree/main/api#apisearch), e.g. test_name.

Reply:
done 👍